examples/community-detection.ipynb

```diff
@@ -0,0 +1,1148 @@
+       "      <td>3712</td>\n",
+       "      <td>[aggies]</td>\n",
+       "      <td>1</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>315</th>\n",
+       "      <td>3759</td>\n",
+       "      <td>[brunei]</td>\n",
+       "      <td>1</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>316</th>\n",
+       "      <td>3769</td>\n",
+       "      <td>[descentintotyranny]</td>\n",
+       "      <td>1</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>317</th>\n",
+       "      <td>3771</td>\n",
+       "      <td>[outofthemetaloop]</td>\n",
+       "      <td>1</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>318</th>\n",
+       "      <td>3773</td>\n",
+       "      <td>[pokemonshowdown]</td>\n",
+       "      <td>1</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "<p>319 rows × 3 columns</p>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "     componentId                                         Subreddits  \\\n",
+       "0              0  [leagueoflegends, nfl, playmygame, dogemarket,...   \n",
+       "1            278  [orangered, orangeredacademy, pasto_range, per...   \n",
+       "2             23  [thedoctorstravels, sirron, aislynisdead, game...   \n",
+       "3            768  [iracing, simracing, redditracing, team_medioc...   \n",
+       "4            832  [perfumeexchange, indiemakeupandmore, asianbea...   \n",
+       "..           ...                                                ...   \n",
+       "314         3712                                           [aggies]   \n",
+       "315         3759                                           [brunei]   \n",
+       "316         3769                               [descentintotyranny]   \n",
+       "317         3771                                 [outofthemetaloop]   \n",
+       "318         3773                                  [pokemonshowdown]   \n",
+       "\n",
+       "     Num_subreddits  \n",
+       "0              3172  \n",
+       "1                20  \n",
+       "2                 8  \n",
+       "3                 6  \n",
+       "4                 6  \n",
+       "..              ...  \n",
+       "314               1  \n",
+       "315               1  \n",
+       "316               1  \n",
+       "317               1  \n",
+       "318               1  \n",
+       "\n",
+       "[319 rows x 3 columns]"
+      ]
+     },
+     "execution_count": 11,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "query = \"\"\"\n",
+    "    CALL gds.wcc.stream('reddit')\n",
+    "    YIELD nodeId, componentId\n",
+    "    RETURN componentId, collect(gds.util.asNode(nodeId).node_id) AS Subreddits, size(collect(gds.util.asNode(nodeId).node_id)) AS Num_subreddits\n",
+    "    ORDER BY size(Subreddits) DESC\n",
+    "\"\"\"\n",
+    "wcc = gds.run_cypher(query)\n",
+    "wcc"
```

Comment:
We already mutated the wcc communities, so we dont need to compute wcc again here.
Its faster to stream the mutated property. See https://neo4j.com/docs/graph-data-science/current/graph-catalog-node-ops/#catalog-graph-stream-single-node-property-example as an example.
Further the RETURN statement can be simplified a bit.
```
    CALL gds.graph.nodeProperties.stream('reddit', 'componentId')
    YIELD nodeId, propertyValue
    WITH nodeId as nodeId, gds.util.asNode(nodeId).node_id AS node, propertyValue AS componentId
    WITH componentId, collect(node) AS nodes, size(collect(nodeId)) AS communitySize
    RETURN componentId, communitySize, nodes
    ORDER BY communitySize DESC
```


Also, I would add an explanation why you use `run_cypher` here instead of the direct gds client call. In this case you want to aggregate the size of the communities.

Reply:
Oh okay! Thank you! :D